client/src/components/home/index.tsx

```diff
@@ -166,10 +176,10 @@ function HomePage(props: {
             variant="extended"
             color="primary"
             disabled={!mentor || isTraining || isLoading || isSaving}
-            onClick={() => startTraining(mentor._id)}
+            onClick={buildAction}
             className={classes.fab}
           >
-            Build Mentor
+            {mentor?.isDirty ? "Build Mentor" : "Preview Mentor"}
```

Comment:
`mentor` has been narrowed above to type `Mentor` (cannot be null or undefined), so everywhere after that narrowing remove the falsy checks on `mentor`